include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -585,17 +585,33 @@ __pattern_reduce_by_segment(_ExecutionPolicy&& __exec, _Range1&& __keys, _Range2
     //          __out_keys   = { 1, 2, 3, 4, 1, 3, 1, 3, 0 }
     //          __out_values = { 1, 2, 3, 4, 2, 6, 2, 6, 0 }
 
-    if (__keys.empty())
+    const auto __n = __keys.size();
+
+    if (__n == 0)
         return 0;
 
+    if (__n == 1)
```

Comment:
Let me check my understanding. The idea of those changes is that previously we copied data from the buffer without calling kernel for `__n == 1` and now we do it all the time except `__n == 0`. Is that right?

I am asking because it's quite hard to figure out what's going on, basing on the PR title

Reply:
Yes, exactly.